numpyro/infer/util.py

```diff
@@ -1008,11 +1008,26 @@ def __call__(self, rng_key, *args, **kwargs):
         elif self.batch_ndims == 1:  # batch over parameters
             batch_size = jnp.shape(tree_flatten(self.params)[0][0])[0]
             rng_keys = random.split(rng_key, batch_size)
+            # TODO: better way to broadcast the model across particles?
```

Comment:
this seems dangerous to me, we define batchsize as the first dimension of one latent variable, if we allow different first dimensions, it is ambiguity which one needs to be batch size.

Reply:
Another solution is to sample the particle assignments first and then treat the predictive distribution as ‘nbatch_dim=0’. I could put the (mixture component) assignment logic in a predictive wrapper under Einstein. The way it doesn't interfere with the other inference methods.